kernel/vmerrors.ml

```diff
@@ -0,0 +1,47 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Pp
+
+type error =
+  | TooLargeInductive of Names.Id.t
+  | TooLargeCode
+
+
+exception CompileError of error
+
+let too_large_code () = raise (CompileError TooLargeCode)
+
+(* Limitation due to OCaml's representation of non-constant
+   constructors: limited to 245 + 1 (0 tag) cases. *)
```

Comment:
This comment no longer makes sense once moved here. Indeed, contrarily to OCaml, Coq's VM is not limited in any way and the limits below (up to 2^24 non-constant constructors) are completely arbitrary.

Reply:
Should we remove the check then?

Reply:
We still need the low-level instructions to work correctly, and SWITCH requires a limit on the number of constructors. Thus, it doesn't hurt to leave this check around, you just need to tweak the comment.